modules/gnome-text-editor/overlay.nix

```
@@ -1,18 +1,24 @@
-{ config, lib, ... }:
+{
+  lib,
+  config,
+  options,
+  ...
+}:
 let
   style = config.lib.stylix.colors {
     template = ../gedit/template.xml.mustache;
     extension = ".xml";
   };
 in
 {
-  options.stylix.targets.gnome-text-editor.enable =
-    config.lib.stylix.mkEnableTarget "GNOME Text Editor" true;
-
   overlay =
     _: prev:
     lib.optionalAttrs
-      (config.stylix.enable && config.stylix.targets.gnome-text-editor.enable)
+      (
+        config.stylix.enable
+        && options.stylix.targets ? gnome-text-editor
+        && config.stylix.targets.gnome-text-editor.enable
```

We could further restrict this with:

```diff
-        && options.stylix.targets ? gnome-text-editor
+        && options.stylix.targets ? gnome-text-editor.enable
```

Otherwise, the better approach would be to use the `or` operator:

```suggestion
        && config.stylix.targets.gnome-text-editor.enable or false
```